application/src/main/java/org/thingsboard/server/service/telemetry/DefaultAlarmSubscriptionService.java

```diff
@@ -235,6 +237,11 @@ private void onAlarmUpdated(AlarmOperationResult result) {
         });
     }
 
+    @Override
+    public List<EntitySubtype> findAlarmTypesByTenantId(TenantId tenantId) {
```

Comment:
Please add AlarmTypeInfo or something similar. EntitySubtype is a redundant structure. You don't have EntityType there. Also, this is a violation of OOP rules. You can't use a class with a completely different meaning.

Reply:
sorry, I missed that ALARM is already part of EntityType. I still believe this is redundant because entityType is always alarm. But we can skip this comment